src/Symfony/Component/Routing/Exception/MissingMandatoryParametersException.php

```diff
@@ -19,4 +19,40 @@
  */
 class MissingMandatoryParametersException extends \InvalidArgumentException implements ExceptionInterface
 {
+    private array $missingParameters = [];
+    private string $routeName = '';
+
+    /**
+     * @param string $routeName
+     * @param array  $missingParameters
+     * @param int    $code
+     */
+    public function __construct(string $routeName = '', $missingParameters = null, /*int*/ $code = 0, \Throwable $previous = null)
+    {
+        if (\is_array($missingParameters)) {
+            $this->missingParameters = $missingParameters;
+            $this->routeName = $routeName;
+            $message = sprintf('Some mandatory parameters are missing ("%s") to generate a URL for route "%s".',
+                implode('", "', $missingParameters),
+                $routeName
+            );
+        } else {
+            trigger_deprecation('symfony/routing', '6.1', 'Construction of "%s" with an exception message is deprecated, provide an array of missing parameters and the route name instead.', __CLASS__);
```

Comment:
the message should now first tell about route name then params I suppose

Reply:
Thanks, fixed.